### PR TITLE
JS Refactoring

### PR DESCRIPTION
**Pull request by callumacrae**
https://tracker.phpbb.com/browse/PHPBB3-13887

### CODE REVIEW

**Comment by callumacrae** [comment]
!set WIP

File: phpBB/styles/prosilver/template/ajax.js [hunk 1]
```diff
@@ -138,9 +138,9 @@ phpbb.markNotifications = function($popup, unreadCount) {
 	}
 
 	// Update page title
-	$('title').text(
-		(unreadCount ? '(' + unreadCount + ')' : '') + $('title').text().replace(/(\(([0-9])\))/, '')
-	);
+	var $title = $('title');
+	var originalTitle = $title.text().replace(/(\((\d+)\))/, '');
+	$title.text((unreadCount ? '(' + unreadCount + ')' : '') + originalTitle);
```
Comment (callumacrae) [inline phpBB/styles/prosilver/template/ajax.js]: I think the code above was buggy; looks like when the number of notifications goes above ten, it doesn't remove the old notification count

**Comment by callumacrae** [comment]
!unset WIP

File: phpBB/styles/prosilver/template/ajax.js (hunk 1 above)
Comment (marc1706) [inline phpBB/styles/prosilver/template/ajax.js]: Yeah, looks like it.

**Comment by nickvergessen** [comment]
3.2? :see_no_evil:

File: phpBB/assets/javascript/core.js [hunk 2]
```diff
@@ -551,11 +538,11 @@ phpbb.search.setValueOnClick = function($input, value, $row, $container) {
  * already, those are displayed instead. Executes the AJAX request function
  * itself due to the need to use a timeout to limit the number of requests.
  *
- * @param array data			Data to be sent to the server.
- * @param object event			Onkeyup event object.
- * @param function sendRequest	Function to execute AJAX request.
+ * @param {Array} data				Data to be sent to the server.
```
Comment (marc1706) [inline phpBB/assets/javascript/core.js]: Should array really be uppercase? I'd go for having it all lowercase

File: phpBB/assets/javascript/core.js [hunk 3]
```diff
@@ -641,12 +626,10 @@ phpbb.search.handleResponse = function(res, $input, fromCache, callback) {
 /**
  * Show search results.
  *
- * @param array results		Search results.
- * @param jQuery $input		Search input|textarea.
- * @param jQuery $container	Search results container element.
- * @param function callback	Optional callback to run when assigning each search result.
- *
- * @return undefined
+ * @param {Array} results		Search results.
```
Comment (marc1706) [inline phpBB/assets/javascript/core.js]: See above comment

**Comment by callumacrae** [comment]
Depends how many merge conflicts you want and how long you want bad code in the core! This code is unlikely to break anything—there's only two changes to logic, and both of them are minor.

File: phpBB/assets/javascript/core.js (hunk 2 above)
Comment (callumacrae) [inline phpBB/assets/javascript/core.js]: Array isn't a type, it's a constructor, so yes, it should be capital.

screen shot 2015-05-29 at 17 47 07

File: phpBB/assets/javascript/plupload.js [hunk 4]
```diff
@@ -595,13 +564,14 @@ phpbb.plupload.uploader.bind('FilesAdded', function(up, files) {
 	}
 
 	// Switch the active tab if the style supports it
-	if (typeof activateSubPanel == 'function') {
-		activateSubPanel('attach-panel');
+	if (typeof activateSubPanel === 'function') {
+		activateSubPanel('attach-panel'); // jshint ignore: line
```
Comment (marc1706) [inline phpBB/assets/javascript/plupload.js]: Comment really needed?

**Comment by marc1706** [comment]
Looks good

File: phpBB/assets/javascript/plupload.js (hunk 4 above)
Comment (callumacrae) [inline phpBB/assets/javascript/plupload.js]: Yep, or jshint throw an error

**Comment by callumacrae** [comment]
Awesome.

@nickvergessen Can we get this in 3.1? It's just refactoring, and I'll probably end up doing it again before 3.2 unless I can get JSHint in the core.

**Comment by marc1706** [comment]
@nickvergessen Please decide on 3.1 or not. To be honest, this is just cosmetics and actually a minor fix for notifications. I'd say this is ok to b added to 3.1.

**Comment by naderman** [comment]
Yeah I'd say 3.1 is fine too

**Pull request merged**